vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomByQrCodeFragment.kt

```diff
@@ -17,39 +17,29 @@
 package im.vector.app.features.createdirect
 
 import android.widget.Toast
-import androidx.annotation.CallSuper
 import com.airbnb.mvrx.activityViewModel
 import com.google.zxing.Result
 import com.google.zxing.ResultMetadataType
 import im.vector.app.R
-import im.vector.app.core.di.ScreenComponent
 import im.vector.app.core.platform.VectorBaseFragment
 import im.vector.app.features.userdirectory.PendingInvitee
 import kotlinx.android.synthetic.main.fragment_qr_code_scanner.*
 import me.dm7.barcodescanner.zxing.ZXingScannerView
-import org.matrix.android.sdk.api.session.Session
 import org.matrix.android.sdk.api.session.permalinks.PermalinkData
 import org.matrix.android.sdk.api.session.permalinks.PermalinkParser
 import org.matrix.android.sdk.api.session.user.model.User
 import javax.inject.Inject
 
 class CreateDirectRoomByQrCodeFragment @Inject constructor() : VectorBaseFragment(), ZXingScannerView.ResultHandler {
 
-    @Inject lateinit var session: Session
+    private val viewModel: CreateDirectRoomViewModel by activityViewModel()
 
     override fun getLayoutResId() = R.layout.fragment_qr_code_scanner
 
-    private val createDirectRoomViewModel: CreateDirectRoomViewModel by activityViewModel()
-
-    @CallSuper
-    override fun injectWith(injector: ScreenComponent) {
-        session = injector.activeSessionHolder().getActiveSession()
-    }
-
     override fun onResume() {
         super.onResume()
         // Register ourselves as a handler for scan results.
-        scannerView.setResultHandler(this)
+        scannerView.setResultHandler(null)
```

Comment:
This is not what you want to do :). Please revert.
What I meant is: `scannerView.setResultHandler(null)` has to be added in the `onPause()` method.

Reply:
Yeah, I've replaced the wrong thing, just noticed that like 10 minutes ago and replaced it, but didn't push yet :P